mongo/change_stream.go

```diff
@@ -367,13 +391,14 @@ AggregateExecuteLoop:
 			// If error is retryable: subtract 1 from retries, redo server selection, checkout
 			// a connection, and restart loop.
 			retries--
-			server, err = cs.client.deployment.SelectServer(ctx, cs.selector)
+			server, err = cs.client.deployment.SelectServer(connCtx, cs.selector)
```

Comment:
All make sense to me since the calculation for selecting a server is based on `min(serverSelectionTimeoutMS, remaining timeoutMS)`. The remaining time being what is left on the context, in this case.

Reply:
Applying the server selection timeout seems like it would apply the calculation
```
min(remaining serverSelectionTimeoutMS, remaining timeoutMS)
```
instead of
```
min(serverSelectionTimeoutMS, remaining timeoutMS)
```

See my proposed timeout pseudocode in [this comment](https://github.com/mongodb/mongo-go-driver/pull/1515#discussion_r1588622132) and let me know if you agree.